src/hardware/input/mouseif_dos_driver.cpp

```diff
@@ -1620,12 +1620,20 @@ static Bitu int33_handler()
 		state.hidden    = 1;
 		// According to Ralf Brown Interrupt List it returns 0x20 if
 		// success,  but CuteMouse source code claims the code for
-		// success if 0x1f. Both agree that 0xffff means failure.
+		// success is 0x1f. Both agree that 0xffff means failure.
 		// Since reg_ax is 0x1f here, no need to change anything.
+		// [FeralChild64] My results:
+		// - MS driver 6.24 always returns 0xffff
+		// - MS driver 8.20 returns 0xffff if 'state.enabled == false'
+		// - 3rd party drivers I tested (A4Tech 8.04a, Genius 9.20,
+		//   Mouse Systems 8.00, DR-DOS driver 1.1) never return anything
 		break;
 	case 0x20: // MS MOUSE v6.0+ - enable mouse driver
 		state.enabled = true;
 		state.hidden  = state.oldhidden;
+		// At least Descent II with Voodoo patch expects this. Checked
+		// that MS driver alters AX this way starting from version 7.
+		reg_ax = 0xffff;
```

Comment:
Note to self: to limit the risk of the change, we should do so in a protected mode only.